### PR TITLE
Display largest number of FT tip amount when there is no aeternity value

### PR DESCRIPTION
**Pull request by stanislav-slavov**

### CODE REVIEW

File: src/components/FiatValue.vue [hunk 1]
```diff
@@ -34,7 +33,10 @@ export default {
     },
     fiatValue() {
       if (!this.rate) return null;
-      if (this.token && !this.tokenData) return null;
+      if (this.token && !this.tokenData) {
+        return new BigNumber(0).toNumber()
+          .toLocaleString('en-US', { style: 'currency', currency: this.showCurrency });
```
Comment (davidyuk) [inline src/components/FiatValue.vue:38]:
```suggestion
        return (0).toLocaleString('en-US', { style: 'currency', currency: this.showCurrency });
```
BigNumber is not needed to do this 🤔

File: src/components/FiatValue.vue [hunk 2]
```diff
@@ -1,6 +1,5 @@
 <template>
   <span
-    v-if="fiatValue"
```
Comment (davidyuk) [inline src/components/FiatValue.vue:3]: Won't it show `~ null` in the case [if `this.rate` is falsy](https://github.com/aeternity/superhero-ui/blob/e7975882f7cf87748520e7d7a84ebf3ba2137d47/src/components/FiatValue.vue#L35)?

File: src/components/TipInput.vue [hunk 3]
```diff
@@ -121,6 +123,25 @@ export default {
         };
       },
     }),
+    largestFtTipAmount() {
+      return this.tipUrlStats.tokenTotalAmount.length
+        ? [...this.tipUrlStats.tokenTotalAmount].sort((a, b) => (
+          shiftDecimalPlaces(new BigNumber(b.amount), -this.tokenInfo[b.token].decimals)
```
Comment (davidyuk) [inline src/components/TipInput.vue:129]: `shiftDecimalPlaces` [already](https://github.com/aeternity/superhero-ui/blob/32076e07b0ac1ad249be984995546e2505fc7890/src/utils/index.js#L7) does the conversion to BigNumber, use one of:
```suggestion
          new BigNumber(b.amount).shiftedBy(-this.tokenInfo[b.token].decimals)
```
or
```suggestion
          shiftDecimalPlaces(b.amount, -this.tokenInfo[b.token].decimals)
```

File: src/components/TipInput.vue [hunk 4]
```diff
@@ -121,6 +123,25 @@ export default {
         };
       },
     }),
+    largestFtTipAmount() {
+      return this.tipUrlStats.tokenTotalAmount.length
+        ? [...this.tipUrlStats.tokenTotalAmount].sort((a, b) => (
```
Comment (davidyuk) [inline src/components/TipInput.vue:128]: Actually, you don't need to sort array to get the max value, I would use `reduce` instead 🤪
```suggestion
        ? this.tipUrlStats.tokenTotalAmount.reduce(
          (a, b) => a.amount > b.amount ? a : b, // use the right comparison from the below
          this.tipUrlStats.tokenTotalAmount[0],
        )
```

File: src/components/TipInput.vue [hunk 5]
```diff
@@ -10,8 +10,9 @@
       <img :src="iconTip">
       <AeAmountFiat
         v-if="!userAddress
-          && (!tipUrlStats.tokenTotalAmount.length || +tipUrlStats.totalAmountAe !== 0)"
-        :amount="tipUrlStats.totalAmountAe"
+          && (tipUrlStats.tokenTotalAmount.length || +tipUrlStats.totalAmountAe !== 0)"
```
Comment (davidyuk) [inline src/components/TipInput.vue:13]: you have already implemented logic to calculate value/token below, why don't reuse it?
```
v-if="!userAddress && tipAmount.value"
```

**Review by mradkov — COMMENTED** [review]
Looks good to me.

**Review by mradkov — APPROVED** [review]

File: src/components/TipInput.vue [hunk 6]
```diff
@@ -121,6 +121,25 @@ export default {
         };
       },
     }),
+    largestFtTipAmount() {
+      return this.tipUrlStats.tokenTotalAmount.length
+        ? this.tipUrlStats.tokenTotalAmount.reduce(
+          (a, b) => (a.amount > b.amount ? a : b),
```
Comment (davidyuk) [inline src/components/TipInput.vue:127]: @stanislav-slavov Shouldn't `a.amount` and `b.amount` be shifted by the decimals value of the corresponding token as it was before?

**Pull request merged**